drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-backend/src/main/java/org/drools/workbench/screens/guided/dtable/backend/server/conversion/SubHeaderBuilder.java

```diff
@@ -105,6 +128,99 @@ public void build() {
         }
     }
 
+    private void makeBRLConditionColumn(final BRLConditionColumn brlColumn) {
+        final Map<String, String> varListInOrder = new HashMap<>();// original, new one
+
+        final GuidedDecisionTable52 dt = new GuidedDecisionTable52();
+        dt.getConditions().add(brlColumn);
+        final ArrayList<DTCellValue52> list = new ArrayList<>();
+        list.add(new DTCellValue52(1));
+        list.add(new DTCellValue52(""));
+        for (int i = 0; i < brlColumn.getChildColumns().size(); i++) {
+            BRLConditionVariableColumn brlConditionVariableColumn = brlColumn.getChildColumns().get(i);
+            if (brlConditionVariableColumn.getVarName().isEmpty()) {
+                list.add(new DTCellValue52(true));
+            } else {
+                String varName = brlConditionVariableColumn.getVarName();
+                String key = "";
+                if (!varName.startsWith("$")) {
+                    key = varName;
+                    varName = "@{" + varName + "}";
+                } else {
+                    key = Integer.toString(i);
+                    varName = "@{" + i + "}";
+                }
+                varListInOrder.put(brlConditionVariableColumn.getVarName(), key);
+                list.add(new DTCellValue52(varName));
+            }
+        }
+        dt.getData().add(list);
+
+        String marshal = subString(GuidedDTDRLPersistence.getInstance().marshal(dt), "when", "then");
+
+        int varIndex = 1;
+        for (BRLConditionVariableColumn childColumn : brlColumn.getChildColumns()) {
+
+            final String var = varListInOrder.get(childColumn.getVarName());
+
+            final Pattern pattern = Pattern.compile("@\\{(" + var + ")}");
+            final Matcher matcher = pattern.matcher(marshal);
+            marshal = matcher.replaceAll("\\$" + varIndex++);
+        }
+
+        makeHeaderAndTitle(CONDITION,
+                           brlColumn.getHeader());
+        fieldRow.createCell(targetColumnIndex).setCellValue(marshal);
+    }
+
+    private String subString(String marshal,
+                             final String from,
+                             final String to) {
+        marshal = marshal.substring(marshal.indexOf(from) + from.length());
+        marshal = marshal.substring(0, marshal.lastIndexOf(to));
+
+        marshal = Arrays.stream(marshal.split("\\r?\\n")).map(String::trim).collect(Collectors.joining("\n"));
+        return marshal;
+    }
```

Comment:
We shouldn't combine two approaches
- updating computed value directly into method argument
- returning computed value as return value of function